test/index.spec.ts

```diff
@@ -1,7 +1,110 @@
-import { hello } from '../src';
+import { GotInstance, GotJSONFn } from 'got';
 
-describe('index.tsx', () => {
-    it('test hello', () => {
-        expect(hello()).toBe('Hello World!');
+import { getSales, createSale } from '../src/index';
+
+declare type ApiInstance = GotInstance<GotJSONFn>;
+
+describe('eNotas API Wrapper', () => {
+  describe('Sales requests', () => {
+    it('GET sales passing query string params', async () => {
+      const qs = {
+        filter: 'data eq "2018-01-01" AND nfe/situacao eq "3"',
+        pageNumber: 0,
+        pageSize: 0,
+        orderBy: 0,
+      };
+
+      const api = ({
+        get: jest.fn(),
+      } as unknown) as ApiInstance;
+
+      await getSales(api)(qs);
+
+      expect(api.get).toHaveBeenCalledWith('/vendas/getFilterBy', {
+        query: qs,
+      });
+    });
+
+    it('POST to create a sale', async () => {
+      const api = ({
+        post: jest.fn(),
+      } as unknown) as ApiInstance;
+
+      await createSale(api)({
```

Comment:
all this guys was required? I think that this kind of test should test only one think:

Throws when something is required
Set default values when we need

Reply:
We can either test it like this or export the mapSale function and write a test for it.. But since mapSale doesn't make sense as a "public" function I think we can keep testing createSale like this.